FIX: BigQuery traceability labels missing in TaskGroup

punitchauhan771 opened the pull request:
<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:


How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

closes: #47297 

This change addresses an issue in the BigQueryInsertJobOperator where labels for task ids inside task groups were not properly formatted. Previously, labels included periods  {taskgroup.task_id} which led to error in storing labels as gcp doesnt support any special character apart from hyphens and underscore in their labels .

reference : 
![image](https://github.com/user-attachments/assets/406e941f-ade8-4b8d-9b55-cd5d72fdb35b)

[supporting_doc](https://cloud.google.com/compute/docs/labeling-resources#:~:text=Requirements%20for%20labels,-The%20labels%20applied&text=Each%20label%20must%20be%20a,maximum%20length%20of%2063%20characters.) 

Now, while storing labels, any period is replaced with a hyphen. As a result, the labels are consistently formatted using the `taskgroup-task_id` format ensuring better traceability and clarity during job executions.

Before:
<img width="926" alt="image" src="https://github.com/user-attachments/assets/ee94eb94-7adf-4604-80ae-9297e19f6dc4" />


After:
<img width="929" alt="image" src="https://github.com/user-attachments/assets/b3b39030-7425-4c26-a7e6-14544ceac125" />

punitchauhan771 committed:
FIX: BigQuery traceability labels missing in TaskGroup

bugraoz93 reviewed (commented):
According to the issue discussion under the issue, it looks good. Could you please include a test case for including a `.` in the label and see if it is without `.` and with `-`?

punitchauhan771 committed:
Test: Add BigQuery TaskGroup traceability label propagation tests

- Added test cases to verify that:
  • Task IDs with dots ('.') are sanitized by replacing them with hyphens ('-').
  • In TaskGroups with prefix_group_id set to False, task IDs containing dots are correctly transformed.
- Removed the obsolete test case from test_labels_invalid_names() that previously raised errors for task names with dots, as this scenario is now supported.

punitchauhan771 committed:
Fix: Replace specific BigQuery query with generic test query

Replaces the hardcoded query referencing `bigquery-public-data.america_health_rankings`
with a generic `SELECT * FROM any` for test purposes. This avoids dependency on a
specific public dataset.

punitchauhan771 committed:
fix: run pre-commit to correct code structure

bugraoz93 reviewed (approved):
Looks good! Thanks for adding the tests! Rebase would be great to keep commits tidy. there are lots of fixes in between. Not a blocker though but good to have
